azure/scope/machine.go

```diff
@@ -323,7 +321,7 @@ func (m *MachineScope) ProviderID() string {
 	if err != nil {
 		return ""
 	}
-	return parsed.ID()
+	return parsed.String()
```

Comment:
I believe this is a long-standing bug, this was returning the VM name instead of the full provider ID (same as the GetVMID() func above)

Reply:
Is there a test that checks this functionality? Does changing this break any other functionality?